src/advanced_unsafety/invalid_values.md

```diff
@@ -0,0 +1,123 @@
+# Invalid values
+
+> _“If you tell the truth, you don't have to remember anything.”__
+>  — _Mark Twain_
+
+Values of a particular type in Rust may never have an "invalid" bit pattern for that type. This is true even if that value is never read from afterwards.
+
+A lot of basic types _don't_ have any rules about invalid values. All bit patterns of the integer types (and arrays of the integer types) are valid.
+
+But most other types have some concept of validity.
```

Comment:
This can be merged with the previous paragraph since it's so short. Alternatively, maybe we could talk about `NonZeroUsize`/`NonNull<T>` as other types that have invalid values?